### PR TITLE
[Enhancement](grouping) Add a switch for users to force using alias name in group by and having clause

### LINKED ISSUE #15747: [Enhancement] Add a switch for users to force using alias name in group by and having clause
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Reference to pr #14408 (issue #14372) and #14077 (issue #14043),
Doris changed the behavior of group by and having clause to be consistent with MySql.
The group by and having clause will use the column name first rather than alias name.

However, users who have been used to the previous behavior react that they have to make a lot of sql changes for this behavior change.

For example, with the following table:

```
mysql> desc table_car;
+-------------+--------------+------+-------+---------+---------+
| Field       | Type         | Null | Key   | Default | Extra   |
+-------------+--------------+------+-------+---------+---------+
| date        | DATE         | Yes  | true  | NULL    |         |
| car_id      | VARCHAR(64)  | Yes  | true  | NULL    |         |
| name        | VARCHAR(200) | Yes  | true  | NULL    |         |
+-------------+--------------+------+-------+---------+---------+
```

The sql will report analysis error:

```
mysql> select CONCAT( car_id, ' ', name ) AS name, count(*) as cnt from table_car group by name;
ERROR 1105 (HY000): errCode = 2, detailMessage = select list expression not produced by aggregation output (missing from GROUP BY clause?): concat(`car_id`, ' ', `name`)
```

After we add a switch to force using alias name in group by and having clause.
It will become:

```
mysql> set group_by_and_having_use_alias_first = true;
Query OK, 0 rows affected (0.00 sec)

mysql> select CONCAT( car_id, ' ', name ) AS name, count(*) as cnt from table_car group by name;
Empty set (0.03 sec)
```

### Solution

Add a switch for users to force using alias name in group by and having clause.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xy720**
# Proposed changes

Issue Number: close #15747 

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.59 seconds
 load time: 492 seconds
 storage size: 17122181433 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230109152006_clickbench_pr_76290.html

**Comment by yangzhg** [comment]
add a regression test

**Comment by morningman** [comment]
Please update variables.md doc

**Comment by xy720** [comment]
Update docs and add regression test

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java [hunk 1]
```diff
@@ -1266,7 +1272,8 @@ private void analyzeAggregation(Analyzer analyzer) throws AnalysisException {
                 groupingInfo.buildRepeat(groupingExprs, groupByClause.getGroupingSetList());
             }
 
-            substituteOrdinalsAliases(groupingExprs, "GROUP BY", analyzer, false);
+            boolean aliasFirst = ConnectContext.get().getSessionVariable().isGroupByAndHavingUseAliasFirst();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java:1275]: The `ConnectContext.get()` may be null when replaying edit log

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**